app/services/project_data_service.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'base64'
+
+class ProjectDataService
+  def initialize(project, access_token)
+    @project = project
+    @client = Octokit::Client.new(access_token: access_token)
+    @circle_config = circle_config
+  end
+
+  def self.refresh_data_for_org(org, access_token)
+    org.projects.each do |project|
+      new(project, access_token).update_computed_properties
+    end
+  end
+
+  def update_computed_properties
+    update_dependencies
+    @project.update(
+      {
+        ci_provider: ci_provider,
+        renovate: renovate?,
+        orbs: orbs
+      }
+    )
+  end
+
+  def ci_provider
+    if @circle_config
+      'circleci'
+    elsif travis?
+      'travis'
+    end
+  end
+
+  def renovate?
+    true if renovate_config
+  end
+
+  def orbs
+    orbs = []
+    @circle_config.include?('artsy/hokusai@') && orbs.push('hokusai')
+    @circle_config.include?('artsy/yarn@') && orbs.push('yarn')
+    orbs
+  end
+
+  def update_dependencies
+    ruby_version && update_dependency('ruby', ruby_version)
+    node_version && update_dependency('node', node_version)
+  end
+
+  def update_dependency(name, version)
+    d = @project.dependencies.where(name: name)
+    if d.empty?
+      @project.dependencies << Dependency.create(name: name, version: version)
+    else
+      d.update(name: name, version: version)
+    end
```

Comment:
Activerecord offers `find_or_initialize_by` which could simplify this method to:

```ruby
@project.dependencies.find_or_initialize_by(name: name).update!(version: version)
```